src/item.cpp

```diff
@@ -6821,6 +6821,11 @@ item_pocket *item::contained_where( const item &contained )
     return contents.contained_where( contained );
 }
 
+const item_pocket *item::contained_where( const item &contained ) const
+{
+    return const_cast<item *>( this )->contained_where( contained );
```

Comment:
I don't think this `const_cast` does anything here. The method is already marked `const` so `this` is `const` already.

Reply:
It's actually casting `this` to a mutable `item *` so that we can use the mutable `contained_where` implementation. We can't call it without the cast, because the mutable version returns a mutable reference, which is a no-go from a const object. It's a common C++ trick to avoid having to duplicate the entire method body for these const/mutable method pairs, and it's safe as long as your mutable version doesn't have side effects.

Reply:
I see; thanks!